manager/controllers/app/fybrikmodule_controller_unit_test.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"context"
+	"testing"
+
+	"fybrik.io/fybrik/manager/controllers/utils"
+
+	app "fybrik.io/fybrik/manager/apis/app/v1alpha1"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// create FybrikModule controller with mockup interfaces
+func createTestFybrikModuleController(cl client.Client, s *runtime.Scheme) *FybrikModuleReconciler {
+	// Create a FybrikModuleReconciler object with the scheme and fake client.
+	return &FybrikModuleReconciler{
+		Client: cl,
+		Name:   "TestModuleReconciler",
+		Log:    ctrl.Log.WithName("test-module-controller"),
+		Scheme: s,
+	}
+}
+
+//nolint:dupl
+func TestFybrikModuleWithInvalidInterface(t *testing.T) {
+	t.Parallel()
+	g := gomega.NewGomegaWithT(t)
+
+	// Set the logger to development mode for verbose logs.
+	logf.SetLogger(zap.New(zap.UseDevMode(true)))
+
+	namespaced := types.NamespacedName{
+		Name:      "module-with-interface-errors",
+		Namespace: "fybrik-system",
+	}
+	filename := "../../testdata/unittests/fybrikmodule-interfaceErrors.yaml"
+	fybrikApp := &app.FybrikModule{}
```

Comment:
change to fybrikModule (all over the file)

Reply:
Done